R/MSnExperiment-class.R

```diff
@@ -146,12 +205,84 @@ readMSnExperiment <- function(file, sampleData, backend = BackendMzR(),
         file, .read_file, files=file, smoothed=smoothed, BPPARAM=BPPARAM
     )))
     backend <- backendInitialize(backend, file, spectraData, BPPARAM=BPPARAM)
-    backend <- backendImportData(backend, file, spectraData, BPPARAM=BPPARAM)
+    backend <- backendImportData(backend, spectraData, BPPARAM=BPPARAM)
     new("MSnExperiment",
         backend = backend,
         sampleData = sampleData,
         spectraData = spectraData,
-        processing = paste0("Data loaded [", date(), "]"),
-        files = file
+        processing = paste0("Data loaded [", date(), "]")
     )
 }
+
+#' @rdname MSnExperiment
+setMethod("spectrapply", "MSnExperiment", function(object, FUN = NULL,
+                                                   BPPARAM = bpparam(), ...) {
+    BPPARAM <- getBpParam(object, BPPARAM = BPPARAM)
+    isOK <- validateFeatureDataForOnDiskMSnExp(object@spectraData)
+    if (length(isOK))
+        stop(isOK)
+    backendSpectrapply(object = object@backend,
+                       spectraData = object@spectraData, FUN = FUN,
+                       BPPARAM = BPPARAM, ...)
+})
+
+#' @rdname MSnExperiment
+setMethod("spectra", "MSnExperiment", function(object, BPPARAM = bpparam())
+    spectrapply(object = object, BPPARAM = BPPARAM))
+
+##============================================================
+##  --  DATA ACCESSORS
+##
+##------------------------------------------------------------
+
+##============================================================
+##  --  SUBSETTING AND FILTERING METHODS
+##
+##------------------------------------------------------------
+
+##============================================================
+##  --  DATA MANIPULATION METHODS
+##
+##------------------------------------------------------------
+
+#' @rdname MSnExperiment
+setMethod("removePeaks", "MSnExperiment", function(object, t = "min",
+                                                   verbose = isMSnbaseVerbose(),
+                                                   msLevel.) {
+    if (!is.numeric(t) & t != "min")
+        stop("Argument 't' has to be either numeric of 'min'.")
+    if (missing(msLevel.))
+        msLevel. <- base::sort(unique(object@spectraData$msLevel))
+    if (!is.numeric(msLevel.))
+        stop("'msLevel' must be numeric.")
+    object@backend <- backendAddProcessing(
+        object@backend, procStep = ProcessingStep("removePeaks",
+                                                  list(t = t, msLevel. = msLevel.)))
+    validObject(object)
```

Comment:
Here you avoid the `if` call already :+1: 
Coding style suggestion: IMHO it is easier to understand if the last lines are:
```r
object@processing <- c(object@processing, ...)
validObject(object)
object
```
`validObject` should be the last check before returning the new object.

Reply:
👍

Reply:
Did that.